File: NoFunctionTest.py
import random

while True:
    try:
        num = float(input("Enter a positive number (or a negative number to exit): "))
        if num < 0:
            print("Exiting program.")
            break
        elif num == 0:
            print("Please enter a number greater than zero.")
            continue
        random_num = random.uniform(0, num)
        print(f"Random number between 0 and {num}: {random_num}")
    except ValueError:
        print("Invalid input. Please enter a valid number.")

File: NoFunctionTest.js
while (true) {
    let input = prompt("Enter a positive number (or a negative number to exit):");
    let num = parseFloat(input);

    if (isNaN(num)) {
        console.log("Invalid input. Please enter a valid number.");
        continue;
    }

    if (num < 0) {
        console.log("Exiting program.");
        break;
    }

    if (num === 0) {
        console.log("Please enter a number greater than zero.");
        continue;
    }

    let randomNum = Math.random() * num;
    console.log(`Random number between 0 and ${num}: ${randomNum}`);
}
